shoop/core/models/_service_behavior.py

```diff
@@ -82,3 +84,70 @@ def get_unavailability_reasons(self, service, source):
         if self.max_weight:
             if weight > self.max_weight:
                 yield ValidationError(_("Maximum weight exceeded."), code="max_weight")
+
+
+class WeightBasedPriceRange(TranslatableModel):
```

Comment:
If these are going to be something that are being offered from the "get go" I'd suggest this would be `PriceRangeService(PolymorphicModel)` or  something like 
```python 
class WeightBasedPriceRange(PriceRangeService):
	unit = "g"``` or something